tests/src/Kernel/AlterableSchemaTest.php

```diff
@@ -0,0 +1,164 @@
+<?php
+
+namespace Drupal\Tests\graphql\Kernel\Framework;
+
+use Drupal\graphql\GraphQL\ResolverRegistry;
+use Drupal\graphql\Plugin\GraphQL\SchemaExtension\SdlSchemaExtensionPluginBase;
+use Drupal\graphql\Plugin\SchemaExtensionPluginManager;
+use Drupal\Tests\graphql\Kernel\GraphQLTestBase;
+use Drupal\Tests\graphql\Kernel\Schema\AlterableComposableTestSchema;
+
+/**
+ * Tests the alterable schema.
+ *
+ * @group graphql
+ */
+class AlterableSchemaTest extends GraphQLTestBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  protected static $modules = [
+    'graphql_alterable_schema_test',
+  ];
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp(): void {
+    parent::setUp();
+    $schema = <<<GQL
+      schema {
+        query: Query
+      }
+      type Query {
+        alterableQuery(id: Int): Result
+      }
+      type Result {
+        id: Int
+      }
+    GQL;
+
+    $this->setUpSchema($schema);
+    $this->mockResolver('Query', 'alterableQuery', function () {
+      return ['id' => 1];
+    });
+  }
+
+  /**
+   * Test if schema altering data is working and arg id is required.
+   */
+  public function testSchemaAlteredQueryArgumentToRequired(): void {
+    $result = $this->query('query { alterableQuery { id } }');
+    $this->assertSame(200, $result->getStatusCode());
+    // The should be error that query argument data is required.
+    $this->assertSame([
+      'errors' => [
+        0 => [
+          'message' => 'Field "alterableQuery" argument "id" of type "Int!" is required but not provided.',
+          'extensions' => [
+            'category' => 'graphql',
+          ],
+          'locations' => [
+            0 => [
+              'line' => 1,
+              'column' => 9,
+            ],
+          ],
+        ],
+      ],
+    ], json_decode($result->getContent(), TRUE));
+  }
+
+  /**
+   * Test if schema extension altering is working and arg position is non-null.
+   */
+  public function testSchemaExtensionAlteredQueryResultPropertyToNonNull(): void {
+    $result = $this->query('query { alterableQuery(id: 1) { id, position } }');
+    $this->assertSame(200, $result->getStatusCode());
+    // The should be error that query argument data position is required.
```

Comment:
Hm, but why do we only see Internal Server error here?

Reply:
Yeah, my bad not a good comment. I think might be better:
```
    // The should be error that query result position variable cannot be null.
    // This leads to the internal server error with reference to the variable.
 ```